### PR TITLE
Adds ETO Dashboard Logic

### PR DESCRIPTION
**Pull request by Moejoe90**
https://github.com/Neufund/platform-frontend/issues/1006
Done 
- Dynamic ETO dashboard logic based on ETO states until `"prospectus_approved"` 
- Adds SubmitEtoWidget 
This Component should enable user to Submit ETO, should only show up when TermSheetWidget is uploaded
- Adds BookBuildingWidget and connected (BookBuilding state) missing starting book building and stopping 
- Adds UploadProspectusWidget, UploadPamphletWidget , UploadTermSheetWidget: 
These are simple components that just check the state of uploaded file and when clicked link to Documents section. 
- Simple Mock etoFile Api
- refactors `ETOFormsProgressSection` removes its connection with state 
- Adds Disable Feature so that you can't access ETO forms if KYC and Email verification is not done
- Initial ChoosePreEtoDateWidget A place holder missing DatePicker component
- Fixes Risk Assesment typo
- Refactors DashboardSection to use `layoutHasDecorator` flag
- wraps  `SocialProfilesEditor` around formWrapper to remove story book error
- Brings back old Selctor In order to calculate 100% of required fields

Missing to be done 
- Add date picker
- Connect ETO to Real file backend when ready
- Connect BookBuilding Widget with Backend when ready
- Add state when ETO is `on-chain`

### CODE REVIEW

File: app/components/eto/EtoDashboard.tsx [hunk 1]
```diff
@@ -1,61 +1,199 @@
...
-};
-
-export const EtoDashboard = appConnect<IStateProps>({
-  stateToProps: s => ({
-    isLightWallet: selectIsLightWallet(s.web3),
-    verifiedEmail: selectVerifiedUserEmail(s.auth),
-    backupCodesVerified: selectBackupCodesVerified(s.auth),
-    requestStatus: selectKycRequestStatus(s.kyc),
+export const EtoDashboard = compose<React.SFC>(
+  appConnect<any, IDispatchProps>({
```
Comment (krzkaczor) [inline app/components/eto/EtoDashboard.tsx]: state props are `any`

File: app/components/eto/registration/pages/RiskAssessment.tsx [hunk 2]
```diff
@@ -31,63 +31,63 @@ type IProps = IStateProps & IDispatchProps;
...
-            label={<FormattedMessage id="eto.form.risk-assesment.no-regulation" />}
+            label={<FormattedMessage id="eto.form.risk-Assessment.no-regulation" />}
           />
         </div>
 
         <div className="form-group">
           <FormCheckbox
             name="riskNoLoansExist"
-            label={<FormattedMessage id="eto.form.risk-assesment.no-loans" />}
+            label={<FormattedMessage id="eto.form.risk-Assessment.no-loans" />}
```
Comment (krzkaczor) [inline app/components/eto/registration/pages/RiskAssessment.tsx]: whats the reasoning for this renaming?

File: app/components/shared/Buttons.tsx [hunk 3]
```diff
@@ -83,3 +84,7 @@ export const ButtonIcon: React.SFC<IButtonIcon> = ({ onClick, className, ...prop
 export const ButtonClose: React.SFC<IGeneralButton> = props => (
   <ButtonIcon {...props} svgIcon={closeIcon} />
 );
+
+export const ButtonArrowRight: React.SFC<any> = props => (
```
Comment (krzkaczor) [inline app/components/shared/Buttons.tsx]: not any but i `IGeneralButton` i guess

File: app/components/eto/registration/pages/RiskAssessment.tsx (hunk 2 above)
Comment (Moejoe90) [reply app/components/eto/registration/pages/RiskAssessment.tsx]: Spell check is wrong so I changed from 
assesment => Assessment

File: app/components/shared/Buttons.tsx (hunk 3 above)
Comment (Moejoe90) [reply app/components/shared/Buttons.tsx]: correct

File: app/components/eto/EtoDashboard.tsx (hunk 1 above)
Comment (Moejoe90) [reply app/components/eto/EtoDashboard.tsx]: done

File: app/components/eto/registration/pages/RiskAssessment.tsx (hunk 2 above)
Comment (krzkaczor) [reply app/components/eto/registration/pages/RiskAssessment.tsx]: Yeah but why is it uppercased?

Comment (Moejoe90) [reply app/components/eto/registration/pages/RiskAssessment.tsx]: changed it to lower case

File: app/modules/eto-flow/selectors.ts [hunk 4]
```diff
@@ -13,6 +16,21 @@ export interface IProgressOptions {
...
+  const maximumErrors = getErrorsNumber(validator, initialFormState.data);
+
+  if (maximumErrors === 0) {
+    return 1;
+  }
+
+  return 1 - errors / maximumErrors;
+};
+
 export type ProgressCalculator = (formState: any, initialData?: any) => number;
```
Comment (trival) [inline app/modules/eto-flow/selectors.ts]: the old calculation was broken. we can use getFormFractionDoneCalculator(Validator, {ignore: true}) To get a progress function, that does not change the fields to required.

File: app/modules/eto-flow/sagas.ts [hunk 5]
```diff
@@ -28,27 +31,49 @@ export function* loadEtoData({ apiEtoService, notificationCenter }: TGlobalDepen
...
+    const currentEtoData = yield effects.select((s: IAppState) => s.etoFlow.etoData);
 
-    const newCompanyData: IHttpResponse<
-      TPartialCompanyEtoData
-    > = yield apiEtoService.putCompanyData({
-      ...oldCompanyData,
+    let newCompanyData: IHttpResponse<TPartialCompanyEtoData> = currentCompanyData;
+    let newEtoData: IHttpResponse<TPartialEtoSpecData> = currentEtoData;
+
+    newCompanyData = yield apiEtoService.putCompanyData({
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/modules/eto-flow/sagas.ts]: I wouldn't say so as what is returned is the response of the backend with the correct data structure as per stored in the backend?

File: app/components/shared/Buttons.tsx [hunk 6]
```diff
@@ -83,3 +84,7 @@ export const ButtonIcon: React.SFC<IButtonIcon> = ({ onClick, className, ...prop
 export const ButtonClose: React.SFC<IGeneralButton> = props => (
   <ButtonIcon {...props} svgIcon={closeIcon} />
 );
+
+export const ButtonArrowRight: React.SFC<IGeneralButton> = props => (
+  <Button layout="secondary" {...props} iconPosition="icon-after" svgIcon={arrowRight} />
+);
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/shared/Buttons.tsx]: done

File: app/modules/eto-flow/sagas.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (trival) [reply app/modules/eto-flow/sagas.ts]: I mean the first one is always overridden. but it is fine with me.

File: app/components/shared/DashboardLinkWidget/DashboardLinkWidget.module.scss [hunk 7]
```diff
@@ -0,0 +1,18 @@
+.content {
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/shared/DashboardLinkWidget/DashboardLinkWidget.module.scss]: This one if you look at the file structure you will see that it's in the shared folder, 
So either the component gets moved to

File: app/components/shared/DashboardLinkWidget/DashboardLinkWidget.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { storiesOf } from "@storybook/react";
+import * as React from "react";
+
+import { DashboardLinkWidget } from "./DashboardLinkWidget";
+
+storiesOf("UploadProspectusWidget", module).add("default", () => (
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/shared/DashboardLinkWidget/DashboardLinkWidget.stories.tsx]: good catch

File: app/components/shared/DashboardLinkWidget/DashboardLinkWidget.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  to: H.LocationDescriptor;
+  buttonText: string | React.ReactNode;
+}
+
+export const DashboardLinkWidget: React.SFC<IProps> = ({ title, text, to, buttonText }) => {
+  return (
+    <Panel headerText={title}>
+      <div className={styles.content}>
+        <p className={cn(styles.text, "pt-2")}>{text}</p>
+        <Col className="d-flex justify-content-center">
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/shared/DashboardLinkWidget/DashboardLinkWidget.tsx]: fair enough

File: app/components/shared/EtoProjectState.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import { storiesOf } from "@storybook/react";
+import * as React from "react";
+
+import { EtoProjectState } from "./EtoProjectStatus";
+
+storiesOf("EtoProjectState", module).add("pending", () => <EtoProjectState status="pending" />);
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/shared/EtoProjectState.stories.tsx]: There are still missing designs for this so i will leave it like this for now

File: app/components/shared/DashboardLinkWidget/DashboardLinkWidget.module.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (trival) [reply app/components/shared/DashboardLinkWidget/DashboardLinkWidget.module.scss]: yes, this one is fine to leave, if it is shared. There were just 3 other similar files in the PR, which can be replace with a shared import. I see you did exactly this, perfect.

**Pull request merged**